classes/local/groups.php

```diff
@@ -73,6 +73,7 @@ public function get_groups() {
 
         try {
             $groups = array();
+            $groups[''] = '';
```

Comment:
I think adding an empty option should be coupled with `mod_form.php`, not this reusable group object, because it's a form-specific behavior. For example, Changing Line#122 in `mod_form.php` to `$templates = array('' => '') + $localgroups->get_groups();`.

Reply:
I added in this method because it's only used for the form as it's not returning all the groups but an array built for the select, however I have moved it to the `mod_form` file in case we need to reuse the method.